### ISSUE TITLE
Question about Markers

### ISSUE THREAD

**Issue by mirecta** [mikes222/mapsforge_flutter#39]
Hi i want add one marker to map but update it periodically (after some data received) how can i do this 
i already had setup map, only live marker missing 
Thx

**Comment by mikes222** [comment]
Create your own implementation of ``IMarkerDataStore``. Add the marker there. Whenever the marker changes call ''notifyListeners()''. Feel free to contact me in case of further questions.

**Comment by mirecta** [comment]
and which marker should i use for drawing little circle

**Comment by mikes222** [comment]
Feel free to create a PR for docu-changes. 
There is no CircleMarker out of the box right now but if you like to you can commit one to the project. The RectMarker could give you a hint how to create one.

**Comment by mikes222** [comment]
Do you have the CircleMarker implemented already? Would be nice to have a PullRequest since I am also needing one.

**Comment by mirecta** [comment]
Yes i have it  but it need little beautify and fix imports to use it inside library 
`import 'package:mapsforge_flutter/marker.dart';
import 'package:mapsforge_flutter/core.dart';
import 'package:mapsforge_flutter/special.dart';
// ignore: implementation_imports
import 'package:mapsforge_flutter/src/graphics/display.dart';
// ignore: implementation_imports
import 'package:mapsforge_flutter/src/model/mappoint.dart';

class CircleMarker<T> extends BasicMarker<T> {
  //static final _log = new Logger('PolygonMarker');

  LatLong center;

  MapPaint? fill;

  double fillWidth;

  int? fillColor;

  MapPaint? stroke;

  final double strokeWidth;

  final int strokeColor;

  final double radius;

  final int? percent;

  CircleMarker({
    display = Display.ALWAYS,
    minZoomLevel = 0,
    maxZoomLevel = 65535,
    item,
    markerCaption,
    required this.center,
    this.radius = 3,
    this.percent,
    this.fillWidth = 1.0,
    this.fillColor,
    this.strokeWidth = 1.0,
    this.strokeColor = 0xff000000,
  })  : assert(display != null),
        assert(minZoomLevel >= 0),
        assert(maxZoomLevel <= 65535),
        assert(strokeWidth >= 0),
        assert(fillWidth >= 0),
        assert(strokeColor != null),
        super(
          display: display,
          minZoomLevel: minZoomLevel,
          maxZoomLevel: maxZoomLevel,
          rotation: 0,
          item: item,
          markerCaption: markerCaption,
        );

  @override
  Future<void> initResources(GraphicFactory graphicFactory) async {
    await super.initResources(graphicFactory);
    if (fill == null && fillColor != null) {
      fill = graphicFactory.createPaint();
      fill!.setColorFromNumber(fillColor!);
      fill!.setStyle(Style.FILL);
      fill!.setStrokeWidth(fillWidth);
      //this.stroke.setTextSize(fontSize);
    }
    if (stroke == null && strokeWidth > 0) {
      stroke = graphicFactory.createPaint();
      stroke!.setColorFromNumber(strokeColor);
      stroke!.setStyle(Style.STROKE);
      stroke!.setStrokeWidth(strokeWidth);
      //this.stroke.setTextSize(fontSize);
    }

    if (markerCaption != null && markerCaption!.latLong == null) {
      markerCaption!.latLong = center;
    }
  }

  @override
  void dispose() {
    super.dispose();
  }

  @override
  bool shouldPaint(BoundingBox? boundary, int zoomLevel) {
    return minZoomLevel <= zoomLevel && maxZoomLevel >= zoomLevel;
  }

  @override
  void renderBitmap(MarkerCallback markerCallback) {
    if (fill != null) {
      markerCallback.renderCircle(
          center.latitude, center.longitude, radius, fill!);
    }
    if (stroke != null) {
      markerCallback.renderCircle(
          center.latitude, center.longitude, radius, stroke!);
    }
  }

  @override
  bool isTapped(
      MapViewPosition mapViewPosition, double tappedX, double tappedY) {
    Mappoint p1 = Mappoint(tappedX + mapViewPosition.leftUpper!.x,
        tappedY + mapViewPosition.leftUpper!.y);
    Mappoint p2 = mapViewPosition.projection!.latLonToPixel(center);

    return p2.distance(p1) >= radius;
  }
}
`

**Comment by mikes222** [comment]
Thx for the code. I have now included the circleMarker along with a unittest.
